src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -761,19 +768,29 @@ class DeclareStatementSegment(BaseSegment):
 
     type = "declare_segment"
     match_grammar = StartsWith("DECLARE")
+    _default_section = Sequence(
+        "DEFAULT",
+        OneOf(
+            Ref("LiteralGrammar"),
+            Bracketed(Ref("SelectStatementSegment")),
+            Ref("BareFunctionSegment"),
+            Ref("FunctionSegment"),
+            Ref("ArrayLiteralSegment"),
+            Ref("TypelessStructSegment"),
+            Ref("TupleSegment"),
+            Ref("BaseExpressionElementGrammar"),
+        ),
+    )
     parse_grammar = Sequence(
         "DECLARE",
         Delimited(Ref("NakedIdentifierSegment")),
-        Ref("DatatypeSegment"),
-        Sequence(
-            "DEFAULT",
-            OneOf(
-                Ref("LiteralGrammar"),
-                Bracketed(Ref("SelectStatementSegment")),
-                Ref("BareFunctionSegment"),
-                Ref("FunctionSegment"),
+        OneOf(
+            Ref("DatatypeSegment"),
+            _default_section,
+            Sequence(
+                Ref("DatatypeSegment"),
+                _default_section,
```

Comment:
@tunetheweb @barrywhart question here around storing some reused grammar in the `_default_section` attribute. The logic is fine, just wondering if you prefer this split into a grammar/segment rather than an attribute for consistency with the rest of the codebase.

A grammar in `bigquery_dialect.add(` (line 79) sounds sensible to me.

Reply:
I prefer a grammar as well. As you say, it's mainly about consistency to avoid confusing future contributors.

Reply:
makes sense!

@KulykDmytro if you take `_default_section` and add it to here: https://github.com/sqlfluff/sqlfluff/blob/bf12e8b2371fdcfc617282256c2d9d2d4d9e9f7e/src/sqlfluff/dialects/dialect_bigquery.py#L79
like:
```
DefaultDeclareOptionsGrammar = Sequence(
        "DEFAULT",
        OneOf(
            Ref("LiteralGrammar"),
            Bracketed(Ref("SelectStatementSegment")),
            Ref("BareFunctionSegment"),
            Ref("FunctionSegment"),
            Ref("ArrayLiteralSegment"),
            Ref("TypelessStructSegment"),
            Ref("TupleSegment"),
            Ref("BaseExpressionElementGrammar"),
        ),
    )
```
and then you can just reference that grammar in your code using `Ref("DefaultDeclareOptionsGrammar")`